modules/core/src/v2/coins/xlm.ts

```diff
@@ -459,52 +460,63 @@ export class Xlm extends BaseCoin {
   }
 
   /**
-   * Attempt to resolve a stellar address into a stellar account
-   * If address domain matches bitgo's then resolve on our federation server
-   * Else, make the request to the federation server hosting the address
+   * Perform federation lookups
+   * Our federation server handles lookups for bitgo as well as for other federation domains
    *
-   * @param address - stellar address to look for
+   * @param {String} [address] - address to look up
+   * @param {String} [accountId] - account id to look up
    */
-  federationLookupByName(address: string): Bluebird<stellar.FederationServer.Record> {
+  private federationLookup({ address, accountId }: { address?: string, accountId?: string }): Bluebird<stellar.FederationServer.Record> {
     const self = this;
     return co<stellar.FederationServer.Record>(function *() {
-      const addressParts = _.split(address, '*');
-      if (addressParts.length !== 2) {
-        throw new InvalidAddressError(`invalid stellar address: ${address}`);
-      }
-      const [, homeDomain] = addressParts;
       try {
-        if (homeDomain === self.homeDomain) {
-          const federationServer = self.getBitGoFederationServer();
+        const federationServer = self.getBitGoFederationServer();
+        if (address) {
           return yield federationServer.resolveAddress(address);
+        } else if (accountId) {
+          return yield federationServer.resolveAccountId(accountId);
         } else {
-          return yield stellar.FederationServer.resolve(address);
+          throw new Error('invalid argument - must provide Stellar address or account id');
         }
       } catch (e) {
-        if (e.message === 'Network Error') {
-          throw e;
+        const error = _.get(e, 'response.data.detail');
+        if (error) {
+          throw new StellarFederationUserNotFoundError(error);
         } else {
-          throw new Error('account not found');
+          throw e;
         }
       }
     }).call(this);
   }
 
+  /**
+   * Attempt to resolve a stellar address into a stellar account
+   *
+   * @param {String} address - stellar address to look for
+   */
+  federationLookupByName(address: string): Bluebird<stellar.FederationServer.Record> {
+    const self = this;
+    if (!address) {
+      throw new Error('invalid Stellar address');
```

Comment:
Nit: Move if block inside coroutine so that this function only has a single error channel. (either throw sync Errors or return rejected promises async , but not both. It makes error handling more complex in some cases).

Reply:
done